src/diffusers/utils/import_utils.py

```diff
@@ -336,6 +336,22 @@ def is_timm_available():
     return _timm_available
 
 
+def is_hpu_available():
+    if (
+        importlib.util.find_spec("habana_frameworks") is None
+        or importlib.util.find_spec("habana_frameworks.torch") is None
+    ):
+        return False
+
+    os.environ["PT_HPU_GPU_MIGRATION"] = "1"
+    logger.debug("Environment variable set: PT_HPU_GPU_MIGRATION=1")
+
+    import habana_frameworks.torch  # noqa: F401
+    import torch
+
+    return hasattr(torch, "hpu") and torch.hpu.is_available()
```

Comment:
```suggestion
```

ohh, the check should just return True or False, indicate if hpu is available or not

Reply:
@yiyixuxu Yes this does return True or False.  We need to set migration to True for HPU before `import habana_frameworks.torch` so this is why I added it as well